.github/workflows/go.yml

```diff
@@ -32,7 +41,46 @@ jobs:
     - name: Build
       run: go build -v .
 
+    - name: Set up testing environment
```

Comment:
Please separate yaml file and let's keep this basic test simple as it is.

Testing is not only to test our source code, but it is also a working example and source of confidence users can use this package on there environment.

When we describe so much steps to setup environment, this could be a message to users, such as, "You need to get this much preparation done to just get started"

Reply:
> Please separate yaml file and let's keep this basic test simple as it is.

As a new workflow, or running a composite action in the `go.yml` workflow?

Example of how composite action would be used:

```diff
      - name: Set up Go
        uses: actions/setup-go@v4
        with:
          go-version: ${{ matrix.go }}
        id: go
      - name: Check out code into the Go module directory
        uses: actions/checkout@v3
      - name: Get dependencies
        run: go get -v -t -d ./...
      - name: Build
        run: go build -v .
+     - name: Set up testing environment
+        uses: ./.github/actions/setup-filesystem
+        id: testenv
      - name: Test
+       working-directory: ${{ testenv.outputs.working_dir }}
+       env:
+         TEST_FILESYSTEM: ${{ matrix.os.filesystem }}
+         TEST_FILECOPYMETHOD: ${{ matrix.os.copymethod }}
        run: go test -v --tags=go${{ matrix.go }}
```

Reply:
New workflow is better

Reply:
Added.